R/git_tools.R

```diff
@@ -48,9 +48,39 @@ checkout_repo <- function(repo_dir, repo_url, select_branch_rule, token_envvar,
       message(paste("clone", repo_url, "to directory", repo_dir))
     }
 
-    git_repo <- git2r::clone(
-      url = repo_url, local_path = repo_dir, credentials = creds, progress = verbose >= 2
-    )
+    tryCatch({
+      git_repo <- git2r::clone(
+        url = repo_url, local_path = repo_dir, credentials = creds, progress = verbose >= 2
+      )
+    }, error = function(e) {
+      # catch some common errors
+      # only do this when cloning because the API calls introduce quite some time overhead
+      if (grepl("https://github.com", repo_url, fixed = TRUE)) {
+        if (!identical(httr::status_code(httr::HEAD("https://github.com")), 200L)) {
+          stop("Host https://github.com not reachable")
+        }
+        repo <- paste(utils::tail(strsplit(repo_url, "/", fixed = TRUE)[[1]], 2), collapse = "/")
+        repo <- substr(repo, start = 0, stop = nchar(repo) - nchar(".git"))
+        tryCatch({
+          # does not work for some reason
+          # httr::HEAD(
+          #   "https://api.github.com/repos/insightsengineering/teal.devel", token = token
+          # )
```

Comment:
would remove if we don't know why it's not working - I wonder if it's a setting in insightsengineering?

Reply:
see comment below, now made it work